src/App.tsx

```diff
@@ -1,18 +1,37 @@
-import React from 'react';
+/* eslint-disable react/jsx-one-expression-per-line */
+/* eslint-disable max-len */
+import React, { useState } from 'react';
 import './App.css';
 import { getNumbers } from './utils';
-
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42)
-  .map(n => `Item ${n}`);
+import { Pagination } from './components/Pagination';
 
 export const App: React.FC = () => {
+  const [currentPage, setCurrentPage] = useState<number>(1);
+  const [perPage, setPerPage] = useState<number>(5);
+
+  const totalItems = 42;
+  const items: string[] = getNumbers(1, totalItems).map((n) => `Item ${n}`);
+  const startIndex: number = (currentPage - 1) * perPage;
+  const endIndex: number = startIndex + perPage;
+  const displayedItems: string[] = items.slice(startIndex, endIndex);
+
+  const handlePageChange = (newPage: number) => {
+    setCurrentPage(newPage);
+  };
+
+  const handlePerPageChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const newPerPage: number = parseInt(e.target.value, 10);
```

Comment:
```suggestion
    const newPerPage = +e.target.value;
```
looks shorter, doesn't it?) (but your option is correct too)